Sources/PackagePlugin/PackageModel.swift

```diff
@@ -206,6 +210,16 @@ public protocol SourceModuleTarget: Target {
     var linkedFrameworks: [String] { get }
 }
 
+/// Represents the kind of module.
+public enum ModuleKind {
+    /// A module that contains generic code (not a test nor an executable).
+    case regular
```

Comment:
is there a more specific term we can use?

Reply:
Nothing comes to mind — do you have any suggestions?  I considered `library` but that's not quite right since non-entry-point modules in an executable aren't really library modules.  Would "other" be too confusing?  (in comparison to "executable" or "test").  This enum case really means "not anything special", so something else that connotes that might make sense.

The other approach I considered was to expose booleans for "contains main entry point" and "contains unit tests", but those are currently always mutually exclusive, and so an enum seemed better.

Reply:
@neonichu any opinions here?

Reply:
Having `other` would at least match the file types, which we have a source, header, resource, or other.

Reply:
I think `regular` seems clear enough considering the context of the other possible values.

Reply:
Maybe we can clarify in the comment (which shows up if you option-click on the enum case in Xcode).

Reply:
@tomerd Happy to change the name here — do you have a particular one in mind?

Reply:
maybe generic is better than regular?

Reply:
Makes sense.  @neonichu do you agree?